ui/src/lib/store/canvasSlice.tsx

```diff
@@ -464,6 +465,70 @@ export const createCanvasSlice: StateCreator<MyState, [], [], CanvasSlice> = (
     }
   },
 
+  importIpynb: (position, repoName, cellList) => {
+    console.log("Sync imported Jupyter notebook cells.");
+    let nodesMap = get().ydoc.getMap<Node>("pods");
+    let scopeNode = createNewNode("SCOPE", position);
+    nodesMap.set(scopeNode.id, scopeNode);
+    get().addPod({
+      id: scopeNode.id,
+      name: repoName,
+      children: [],
+      parent: "ROOT",
+      type: scopeNode.type as "CODE" | "SCOPE" | "RICH",
+      lang: "python",
+      x: scopeNode.position.x,
+      y: scopeNode.position.y,
+      width: scopeNode.width!,
+      height: scopeNode.height!,
+      // For my local update, set dirty to true to push to DB.
+      dirty: true,
+      pending: true,
+    });
+    if (cellList.length > 0) {
+      for (let i = 0; i < cellList.length; i++) {
+        const cell = cellList[i];
+        let newPos = {
+          x: position.x + 50,
+          y: position.y + 100 + i * 150,
+        };
+
+        let node = createNewNode(
+          cell.cellType == "code" ? "CODE" : "RICH",
+          newPos
+        );
+        let podContent = cell.cellType == "code" ? cell.cellSource : "";
+        let podRichContent = cell.cellType == "markdown" ? cell.cellSource : "";
+
+        nodesMap.set(node.id, node);
+        get().addPod({
+          id: node.id,
+          children: [],
+          parent: scopeNode.id,
+          type: node.type as "CODE" | "SCOPE" | "RICH",
+          lang: "python",
+          x: node.position.x,
+          y: node.position.y,
+          width: node.width!,
+          height: node.height!,
+          content: podContent,
+          richContent: podRichContent,
+          // For my local update, set dirty to true to push to DB.
+          dirty: true,
+          pending: true,
+        });
+        get().moveIntoScope(node.id, scopeNode.id);
```

Comment:
The current `moveIntoScope` function is pretty heavy because it calls the three heavy functions `adjustLevel()`, `updateView`, and `buildNode2Children` at its end ([ref](https://github.com/codepod-io/codepod/blob/af834f8a8f253ccf18a3a85bfffa41a375cb6d26/ui/src/lib/store/canvasSlice.tsx#L710-L713)). I can see it to be very slow when a repo becomes large.

Those three functions only need to be called once at the end of the ipynbImport action. So let's add a `moveIntoScopeLight` without those three functions and use it here.

```ts
moveIntoScopeLight() {...}
moveIntoScope() {
    moveIntoScopeLight(); 
    get().adjustLevel();
    get().updateView();
    get().buildNode2Children();
}
importIpynb() {
    ...
    for() {...
        moveIntoScopeLight()
    }
    get().adjustLevel();
    get().updateView();
    get().buildNode2Children();
    ...
}
```